### PR TITLE
feat: parsing errors page

### LINKED ISSUE #155: Display parsing errors in interface

### PR DESCRIPTION
**Pull request by diivi**
* Added a page to show all errors related to a taxonomy
* Added the Errors column to /gotoproject table
image
image
image

FIxes #155

### CODE REVIEW

File: taxonomy-editor-frontend/src/pages/gotoproject/index.jsx [hunk 1]
```diff
@@ -65,7 +66,22 @@ const GotoProject = () => {
...
+            {
+              title: "Errors",
+              field: "errors_count",
+              render: (rowData) => {
+                return (
+                  <Typography
+                    color={rowData["errors_count"] > 0 ? "error" : "success"}
+                  >
+                    {rowData["errors_count"] > 0
+                      ? rowData["errors_count"] + " errors"
```
Comment (alexgarel) [inline taxonomy-editor-frontend/src/pages/gotoproject/index.jsx:79]: Could it be a link to this project errors page ?

File: taxonomy-editor-frontend/src/pages/search/index.jsx [hunk 2]
```diff
@@ -24,6 +24,7 @@ const SearchNode = ({ setDisplayedPages }) => {
       setDisplayedPages([
         { url: urlPrefix + "entry", translationKey: "Nodes" },
         { url: urlPrefix + "search", translationKey: "Search" },
+        { url: urlPrefix + "errors", translationKey: "Errors" },
```
Comment (alexgarel) [inline taxonomy-editor-frontend/src/pages/search/index.jsx:27]: Waow, on each menu change we edit the same code every where.

It's for another PR, but shouldn't this menu be a component per se ? (used on every page)

**Review by alexgarel — COMMENTED** [review]
I'll let @nobeeakon do the final review :-)

Just have one comment on adding a link

**Comment by alexgarel** [comment]
And thank you so much @diivi

File: taxonomy-editor-frontend/src/pages/search/index.jsx (hunk 2 above)
Comment (diivi) [reply taxonomy-editor-frontend/src/pages/search/index.jsx:27]: Yeah, seemed redundant to me too, I'll create an issue after this one.

Comment (nobeeakon) [reply taxonomy-editor-frontend/src/pages/search/index.jsx:27]: yep, I was planning to take that one

File: taxonomy-editor-frontend/src/pages/errors/index.jsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    isPending,
+    isError,
+    __isSuccess,
+    errorMessage,
+  } = useFetch(`${baseUrl}parsing_errors`);
+
+  useEffect(() => {
+    if (!errorData) return;
+    errorData.errors.forEach((error, index) => {
+      setErrors((prevErrors) => [
```
Comment (nobeeakon) [inline taxonomy-editor-frontend/src/pages/errors/index.jsx:35]: what about doing a `.map` and 1 single `setErrors`

File: taxonomy-editor-frontend/src/pages/gotoproject/index.jsx [hunk 4]
```diff
@@ -65,7 +66,25 @@ const GotoProject = () => {
...
+                  <MuiLink
+                    color="error"
+                    href={`/${toSnakeCase(rowData["taxonomyName"])}/${
+                      rowData["branchName"]
+                    }/errors`}
+                  >
+                    {rowData["errors_count"] + " errors"}
+                  </MuiLink>
+                ) : (
+                  <Typography color="success">✅</Typography>
```
Comment (nobeeakon) [inline taxonomy-editor-frontend/src/pages/gotoproject/index.jsx:84]: think is ok to leave it empty if no error

File: taxonomy-editor-frontend/src/pages/errors/index.jsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+  const title = toTitleCase(taxonomyName);
+  const baseUrl = createBaseURL(taxonomyName, branchName);
+  const urlPrefix = `${taxonomyName}/${branchName}/`;
+  const [errors, setErrors] = useState([]);
+  /* eslint no-unused-vars: ["error", { varsIgnorePattern: "^__" }] */
+  const {
+    data: errorData,
+    isPending,
+    isError,
+    __isSuccess,
```
Comment (nobeeakon) [inline taxonomy-editor-frontend/src/pages/errors/index.jsx:28]: nit: you can remove this one as is not used

File: taxonomy-editor-frontend/src/pages/errors/index.jsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+          id: index + 1,
+          error: error,
+        },
+      ]);
+    });
+  }, [errorData]);
+
+  // Set url prefix for navbar component
+  useEffect(
+    function addUrlPrefixToNavbar() {
```
Comment (nobeeakon) [inline taxonomy-editor-frontend/src/pages/errors/index.jsx:47]: nit: please rename to `addLinksToNavbar`, also please update the comment

File: taxonomy-editor-frontend/src/pages/errors/index.jsx [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+import MaterialTable from "@material-table/core";
+import { useState, useEffect } from "react";
+
+const Errors = ({ setDisplayedPages }) => {
+  const { taxonomyName, branchName } = useParams();
+  const title = toTitleCase(taxonomyName);
+  const baseUrl = createBaseURL(taxonomyName, branchName);
+  const urlPrefix = `${taxonomyName}/${branchName}/`;
+  const [errors, setErrors] = useState([]);
+  /* eslint no-unused-vars: ["error", { varsIgnorePattern: "^__" }] */
```
Comment (nobeeakon) [inline taxonomy-editor-frontend/src/pages/errors/index.jsx:23]: please do avoid disabling eslint

File: taxonomy-editor-frontend/src/pages/errors/index.jsx [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+            <TableCell align="left">
+              <Typography variant="h6">Taxonony Name</Typography>
+            </TableCell>
+            <TableCell align="left">
+              <Typography variant="h6">Branch Name</Typography>
+            </TableCell>
+          </TableHead>
+          <TableBody>
+            <TableCell align="left" component="td" scope="row">
+              <Typography variant="body1">{title}</Typography>
```
Comment (nobeeakon) [inline taxonomy-editor-frontend/src/pages/errors/index.jsx:87]: `title`  seems to be used only once, think is ok not to have it as a variable ... ` <Typography variant="body1">{toTitleCase(taxonomyName)}</Typography>`

File: taxonomy-editor-frontend/src/pages/gotoproject/index.jsx (hunk 4 above)
Comment (diivi) [reply taxonomy-editor-frontend/src/pages/gotoproject/index.jsx:84]: No feedback seemed a little weird, so I added a hyphen here

File: taxonomy-editor-frontend/src/pages/errors/index.jsx [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+  const {
+    data: errorData,
+    isPending,
+    isError,
+    errorMessage,
+  } = useFetch(`${baseUrl}parsing_errors`);
+
+  useEffect(() => {
+    if (!errorData) return;
+    const errors = [];
```
Comment (nobeeakon) [inline taxonomy-editor-frontend/src/pages/errors/index.jsx:31]: `   const newErrors = errorData.errors.map((error, index) => ({ id: index + 1, error: error })); `

File: taxonomy-editor-frontend/src/pages/errors/index.jsx [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+        { url: urlPrefix + "export", translationKey: "Export" },
+      ]);
+    },
+    [urlPrefix, setDisplayedPages]
+  );
+
+  if (isError) {
+    return (
+      <Box>
+        <Typography variant="h5">{errorMessage}</Typography>
```
Comment (nobeeakon) [inline taxonomy-editor-frontend/src/pages/errors/index.jsx:53]: Let's just show: 'Unable to load errors' or something

File: taxonomy-editor-frontend/src/pages/gotoproject/index.jsx [hunk 11]
```diff
@@ -65,7 +66,25 @@ const GotoProject = () => {
...
+                  <MuiLink
+                    color="error"
+                    href={`/${toSnakeCase(rowData["taxonomyName"])}/${
+                      rowData["branchName"]
+                    }/errors`}
+                  >
+                    {rowData["errors_count"] + " errors"}
+                  </MuiLink>
+                ) : (
+                  <Typography>-</Typography>
```
Comment (nobeeakon) [inline taxonomy-editor-frontend/src/pages/gotoproject/index.jsx:84]: think is ok to not show anything.

**Review by nobeeakon — APPROVED** [review]
well done!!

**Pull request merged**